lib/capistrano/tasks/deploy.rake

```diff
@@ -149,9 +149,12 @@ namespace :deploy do
   task :cleanup do
     on release_roles :all do |host|
       releases = capture(:ls, "-x", releases_path).split
-      if !(releases.all? { |e| /^\d{14}$/ =~ e })
-        warn t(:skip_cleanup, host: host.to_s)
-      elsif releases.count >= fetch(:keep_releases)
+      invalid_releases = releases.select { |e| /^\d{14}$/ =~ e }
```

Comment:
this should be `reject` instead. you're taking out the actual releases here and it will just result in not rotation again.